src/suspendplugin.c

```diff
@@ -441,9 +491,16 @@ void
 suspend_plugin_exit()
 {
     DBG("");
+
     mce_display_remove_all_handlers(mce_display, mce_display_event_ids);
     mce_display_unref(mce_display);
     mce_display = NULL;
+
+    g_io_channel_shutdown(netlink_channel, TRUE, NULL);
+    g_io_channel_unref(netlink_channel);
+
+    nl_socket_free(nl_socket);
```

Comment:
g_io_channel_unref() closes the socket fd ->  nl_socket_free() operates on stale fd?

Reply:
Yes, the socket is closed twice. First by `g_io_channel_shutdown()`:
```
#0  close () at ../sysdeps/unix/syscall-template.S:82
#1  0xf730d95a in g_io_unix_close (channel=<optimized out>, err=0x0) at giounix.c:320
#2  0xf72ba7da in g_io_channel_shutdown (channel=0x147bc0, flush=<optimized out>, err=0x0) at giochannel.c:523
#3  0xf77e53f8 in suspend_plugin_exit () at src/suspendplugin.c:499
#4  0x000556f8 in __connman_plugin_cleanup () at src/plugin.c:242
#5  0x00053d30 in main (argc=1, argv=0xfffefb14) at src/main.c:905
```
and then by `nl_socket_free()`:
```
#0  close () at ../sysdeps/unix/syscall-template.S:82
#1  0xf6df2420 in nl_socket_free (sk=0x13cf48) at lib/socket.c:249
#2  0xf77e5420 in suspend_plugin_exit () at src/suspendplugin.c:502
#3  0x000556f8 in __connman_plugin_cleanup () at src/plugin.c:242
#4  0x00053d30 in main (argc=1, argv=0xfffefb14) at src/main.c:905
```
The second `close()` returns -1.